openedx/openedx-platform
# feat: update tab_handler to include blockId

## Description

This PR implements the json request section of the `tab_handler` API. The API will return a JSON object when the request comes from a JSON application. Additionally this PR adds page redirect to the course authoring MFE if the corresponding WaffleFlag is active. This change impacts Developers and Authors.

## Supporting information

JIRA Ticket: [TNL-10795](https://2u-internal.atlassian.net/browse/TNL-10795)

## Testing instructions

1.  Enable `contentstore.new_studio_mfe.use_new_custom_pages` and `discussions.pages-and-resources`
2. Navigate to "Pages and Resources"
3. Click on "Custom Pages"
4. Render expected custom pages.
5. Click "Add a new page"
6. New page should be added
7. Click delete icon and confirm
8. Corresponding page should be deleted

## Deadline

None

## Other information

Need to `course-authoring` PR [#510](https://github.com/openedx/frontend-app-course-authoring/pull/510) for testing
